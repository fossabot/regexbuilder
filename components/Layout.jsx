// const DefaultLayout = React.createClass({
//   propTypes: {
//     title: React.PropTypes.string
//   },
//   render() {
//     return (
//       <div>
//         <div className="l-header">
//           <h1>Company Name</h1>
//           <Navigation />
//         </div>
//         <div className="l-content">
//           <h1>{this.props.title}</h1>
//           {this.props.children}
//         </div>
//         <div className="l-footer">
//           <p>Copyright (c) 2014 Company Name</p>
//         </div>
//       </div>
//     );
//   },
//   componentDidMount() {
//     if (ExecutionEnvironment.canUseDOM) {
//       document.title = this.props.title;
//     }
//   }
// });

import React from "react";

function Layout() {
  return (
    <div>
      <div className="l-header">
        <h1>Company Name</h1>
      </div>
      <div className="l-content">
        <h1>NICE</h1>
      </div>
      <div className="l-footer">
        <p>Copyright (c) 2014 Company Name</p>
      </div>
    </div>
  );
}

export default Layout;
