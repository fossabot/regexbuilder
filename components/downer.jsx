import React from "react";

function Downer() {
  return (
    <header>
      <h1>Next.js Example on Now 2.0</h1>
    </header>
  );
}

export default Downer;
