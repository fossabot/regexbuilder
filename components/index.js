export { default as Header } from "./header";
export { default as Downer } from "./downer";
