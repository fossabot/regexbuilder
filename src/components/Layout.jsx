import React from 'react';
import Head from 'next/head';

export default function Layout() {
  return (
    <div>
      <Head>
        <title>NNN</title>
        <link
          rel='icon'
          href='/static/images/favicon.ico'
          type='image/x-icon'
        />
      </Head>
      <div className='l-header'>
        <h1>Company Name</h1>
      </div>
      <div className='l-content'>
        <h1>NICE</h1>
      </div>
      <div className='l-footer'>
        <p>Copyright (c) 2014 Company Name</p>
      </div>
    </div>
  );
}
