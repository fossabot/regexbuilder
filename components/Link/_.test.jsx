import React from 'react';
import renderer from 'react-test-renderer';
import { Link } from 'components';

test('Link should changes the class when hovered and no href is supplied', () => {
  const component = renderer.create(<Link>Test</Link>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.onMouseEnter();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.onMouseLeave();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('Link should changes the class when hovered and if a href is supplied', () => {
  const ref = 'https://youtube.com';
  const component = renderer.create(<Link href={ref}>Test</Link>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.onMouseEnter();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.onMouseLeave();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('should set href to "#" if no href is set', () => {
  const component = renderer.create(<Link>Test</Link>);
  expect(component.root.props.href).toBe('#');
});

test('should set href to supplied href', () => {
  const ref = 'https://youtube.com';
  const component = renderer.create(<Link href={ref}>Test</Link>);
  expect(component.root.props.href).toBe(ref);
});
