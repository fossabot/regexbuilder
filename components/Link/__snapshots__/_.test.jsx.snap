// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should changes the class when hovered and if a href is supplied 1`] = `
<a
  className="normal"
  href="https://youtube.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Test
</a>
`;

exports[`Link should changes the class when hovered and if a href is supplied 2`] = `
<a
  className="hovered"
  href="https://youtube.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Test
</a>
`;

exports[`Link should changes the class when hovered and if a href is supplied 3`] = `
<a
  className="normal"
  href="https://youtube.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Test
</a>
`;

exports[`Link should changes the class when hovered and no href is supplied 1`] = `
<a
  className="normal"
  href="#"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Test
</a>
`;

exports[`Link should changes the class when hovered and no href is supplied 2`] = `
<a
  className="hovered"
  href="#"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Test
</a>
`;

exports[`Link should changes the class when hovered and no href is supplied 3`] = `
<a
  className="normal"
  href="#"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Test
</a>
`;
